### PR TITLE
Incorporate haul into custom function project to allow for node-debugging

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
This PR enables node-debugging for custom functions outside Fabric Internal. The contents of the "haul folder eventually need to be packaged up so that we can add the package as a dependency to the custom functions project.

I still need to figure out if it's possible to reference the bundle as opposed to the source files in haul.config.js and then have debugging work. I found that if I reference the bundle in haul.config.js, I can debug the bundle but not the individual .js file, which is not ideal

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
```
Comment (akrantz) [inline .vscode/launch.json]: Please update this to match the latest in excel-functions in fabric-internal.

File: package.json [hunk 2]
```diff
@@ -3,18 +3,30 @@
   "version": "2.0.0",
   "main": "./src/customfunctions.js",
   "scripts": {
-    "start": "start npm run dev-server && start npm run watch && npm run sideload",
+    "start": "start npm run packager && start npm run dev-server && start npm run watch && npm run sideload",
```
Comment (akrantz) [inline package.json]: Please update this based on excel-functions in fabric-internal.

File: package.json [hunk 3]
```diff
@@ -3,18 +3,30 @@
...
     "build": "webpack -p --mode production",
-    "dev-server": "webpack-dev-server --mode development --https --key ./certs/server.key --cert ./certs/server.crt --cacert ./certs/ca.crt --port 8081 --hotOnly",
+    "dev-server": "webpack-dev-server --mode development --https --key ./certs/server.key --cert ./certs/server.crt --cacert ./certs/ca.crt --port 8080 --hotOnly",
+    "enable-debugging": "office-addin-dev-settings enable-debugging manifest.xml && office-addin-dev-settings set-source-bundle-url manifest.xml --path index.win32",
+    "packager": "haul start --config ./packager/haul.config.js",
     "sideload": "office-toolbox sideload -m manifest.xml -a excel",
     "watch": "webpack -p --mode development --watch"
   },
   "devDependencies": {
+    "@types/es6-collections": "^0.5.29",
```
Comment (akrantz) [inline package.json]: Need to make the packager separate so we don't have the react-native references.

File: packager/haul.config.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      //
+      const initializeCoreLocation = relative(
+        process.cwd(),
+        require.resolve('./platform-bundle-initcore-placeholder.js')
+      );
+
+      //  webpack configuration        
+      const factory = createWebpackConfig({
+        ...options,
+        entry: { index: [`./src/customfunctions.js`] },
```
Comment (akrantz) [inline packager/haul.config.js]: Needs to work with the bundle, and we need source maps working.

File: packager/platform-bundle-bootstrapper.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+if (global.__platformBundles != undefined) {
```
Comment (akrantz) [inline packager/platform-bundle-bootstrapper.js]: We don't need the platform-bundle-bootstrapper. office-addin-node-debugger supports this.

File: packager/haul.config.js (hunk 4 above)
Comment (TCourtneyOwen) [reply packager/haul.config.js]: As I mentioned, "I still need to figure out if it's possible to reference the bundle as opposed to the source files in haul.config.js and then have debugging work. I found that if I reference the bundle in haul.config.js, I can debug the bundle but not the individual .js file, which is not ideal"

File: index.html [hunk 6]
```diff
@@ -6,13 +6,13 @@
   <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
   <meta http-equiv="Expires" content="0" />
   <title></title>
-  <script src="https://unpkg.com/@microsoft/office-js@custom-functions-preview-refresh/dist/custom-functions-runtime.js" type="text/javascript"></script>
-  <script src="dist/win32/ship/index.win32.bundle" type="text/javascript"></script>
+  <script src="https://officedev.github.io/custom-functions/lib/custom-functions-runtime.js" type="text/javascript"></script>
+  <script src="dist/win32/ship/index.win32.bundle.js" type="text/javascript"></script>
```
Comment (Zlatkovsky) [inline index.html]: We need to use the web vs. native bundles, like I have in my reworked templates.  It makes no sense to have "win32" in here, esp. since this file only works on the WAC!

File: manifest.xml [hunk 7]
```diff
@@ -14,7 +14,7 @@
 		<Host Name="Workbook" />
 	</Hosts>
 	<DefaultSettings>
-		<SourceLocation DefaultValue="https://localhost:8081/index.html"/>
+		<SourceLocation DefaultValue="https://localhost:8080/index.html"/>
```
Comment (Zlatkovsky) [inline manifest.xml]: Why the change in port?

File: package.json (hunk 2 above)
Comment (Zlatkovsky) [reply package.json]: Should we have "start" only do the devserver and watch?  (esp. as our recommendation, for now, is to always start with just getting it to work on the web)?  I.e., replace "start" with what we have for "start-web".

And then have start-desktop do the sideloading (or keep them as separate commands, that's ok too).

File: manifest.xml (hunk 7 above)
Comment (akrantz) [reply manifest.xml]: Because it is being served from the dev-server instead of the packager.

**Comment by akrantz** [comment]
@Zlatkovsky You should look at the excel-functions project in fabric-internal. It supports web and desktop debugging, and the readme is documented. If there are changes that you would like to see, please provide a list of items so we can be sure to address them. For the "npm start", I've made it so that you can edit that to use "npm start-web", "npm start-desktop" or "npm start-desktop-direct" for Excel Online, Desktop, and Desktop with direct debugging (respectively). There is also "npm stop". These things need to be ported over to this repository but there has been so much churn, and we're still working on running the packager outside fabric-internal, so it has been hard to also address the things you want to see. I hope you understand that we appreciate your input but we need to be organized about how go about completing this.

**Comment by akrantz** [comment]
Incorporated into https://github.com/OfficeDev/Excel-Custom-Functions/pull/54.
